cmake/sysbuild/image_signing_extra.cmake

```diff
@@ -0,0 +1,132 @@
+  return()
+endif()
+
+set(keyfile "${SB_CONFIG_BOOT_SIGNATURE_KEY_FILE}")
+set(keyfile_enc "${SB_CONFIG_BOOT_ENCRYPTION_KEY_FILE}")
+string(CONFIGURE "${keyfile}" keyfile)
+string(CONFIGURE "${keyfile_enc}" keyfile_enc)
+
+if(NOT "${SB_CONFIG_BOOT_SIGNATURE_TYPE_NONE}")
+  if("${keyfile}" STREQUAL "")
+    message(WARNING "Neither SB_CONFIG_BOOT_SIGNATURE_TYPE_NONE or "
+                    "SB_CONFIG_BOOT_SIGNATURE_KEY_FILE are set, the generated build will not be "
+                    "bootable by MCUboot unless it is signed manually/externally.")
+    return()
+  endif()
+endif()
+
+if(NOT DEFINED IMGTOOL)
+  message(FATAL_ERROR "Can't sign images for MCUboot: can't find imgtool. To fix, install imgtool with pip3, or add the mcuboot repository to the west manifest and ensure it has a scripts/imgtool.py file.")
+  return()
+endif()
+
+get_property(extra_names GLOBAL PROPERTY DFU_EXTRA_NAMES)
+get_property(extra_versions GLOBAL PROPERTY DFU_EXTRA_VERSIONS)
+get_property(extra_targets GLOBAL PROPERTY DFU_EXTRA_TARGETS)
+
+# Iterate through all extra images and sign them
+list(LENGTH extra_paths num_binaries)
+if(num_binaries GREATER 0)
+  math(EXPR last_index "${num_binaries} - 1")
+
+  set(signed_paths "")
+  set(signed_names "")
+
+  foreach(index RANGE ${last_index})
+    math(EXPR slot_number "${index} + 1")
+
+    if(NOT DEFINED NCS_MCUBOOT_EXTRA_${slot_number}_IMAGE_NUMBER OR NCS_MCUBOOT_EXTRA_${slot_number}_IMAGE_NUMBER EQUAL -1)
+      message(FATAL_ERROR "Extra image slot ${slot_number} not configured. Set SB_CONFIG_MCUBOOT_EXTRA_IMAGES to at least ${num_binaries} in sysbuild.")
+    endif()
+
+    set(image_id ${NCS_MCUBOOT_EXTRA_${slot_number}_IMAGE_NUMBER})
+
+    list(GET extra_paths ${index} input_bin)
+    list(GET extra_names ${index} image_name)
+    list(GET extra_versions ${index} version)
+
+    if(NOT image_name OR image_name STREQUAL "")
+      get_filename_component(image_name "${input_bin}" NAME_WE)
+    endif()
+
+    set(output_bin "${CMAKE_BINARY_DIR}/${image_name}.signed.bin")
+    set(output_hex "${CMAKE_BINARY_DIR}/${image_name}.signed.hex")
+
+    set(imgtool_sign 
+      ${PYTHON_EXECUTABLE} ${IMGTOOL} sign
+      --version ${version}
+      --align 16
+      --slot-size $<TARGET_PROPERTY:partition_manager,PM_MCUBOOT_PRIMARY_${image_id}_SIZE>
+      --pad-header
+      --header-size ${SB_CONFIG_PM_MCUBOOT_PAD}
+    )
+
+    set(imgtool_extra "")
+    if(NOT "${keyfile}" STREQUAL "")
+      list(APPEND imgtool_extra -k "${keyfile}")
+    endif()
+
+    # Add SHA algorithm based on MCUboot configuration (same as image_signing.cmake)
+    sysbuild_get(CONFIG_MCUBOOT_BOOTLOADER_USES_SHA512 IMAGE mcuboot VAR CONFIG_MCUBOOT_BOOTLOADER_USES_SHA512 KCONFIG)
+    if(CONFIG_MCUBOOT_BOOTLOADER_USES_SHA512)
+      list(APPEND imgtool_extra --sha 512)
+    endif()
+
+    sysbuild_get(CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION IMAGE ${DEFAULT_IMAGE} VAR CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION KCONFIG)
```

Comment:
sysbuild has it, `SB_CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION` but not sure if it's needed here anyway since it's only used for 1 image iirc, @de-nordic for comment, can probably leave it in though anyway